### ISSUE TITLE
Change signature of `option` to use an object

### ISSUE THREAD

**Issue by eschwartz** [four43/admiral-cli#7]
The signature of `addOption` is getting a little lengthy... and I think there are more options that should be available. How 'bout moving the parameters to an options object:

```js
cli.option('someOption', {
  description: 'Options are two parts, a key and a user supplied value',
  flag: '--some-option',
  shortFlag: '-o',
  type: 'string',
  length: 1
});
```

**Comment by four43** [comment]
Agreed, they are getting muddled the more we add to them. Looks like a nice breaking change to have before going v1.0.0

**Comment by four43** [comment]
This has been fixed on dev, if you'd like to take a look. Going to make sure NPM won't pull these braking changes, then I can merge in to master. Thanks @eschwartz

**Comment by four43** [comment]
Fixed via #12

**Comment by four43** [comment]
Fixed via #12
